polonius-engine/src/output/mod.rs

```diff
@@ -213,22 +223,67 @@ impl<T: FactTypes> Output<T> {
 
         let killed = all_facts.killed.clone().into();
 
+        // `known_subset` is a list of all the `'a: 'b` subset relations the user gave:
+        // it's not required to be transitive. `known_contains` is its transitive closure: a list
+        // of all the known placeholder loans that each of these placeholder origins contains.
+        // Given the `known_subset`s `'a: 'b` and `'b: 'c`: in the `known_contains` relation, `'a`
+        // will also contain `'c`'s placeholder loan.
+        let known_subset = all_facts.known_subset.clone().into();
+        let known_contains =
+            Output::<T>::compute_known_contains(&known_subset, &all_facts.placeholder);
+
+        let placeholder_origin: Relation<_> = Relation::from_iter(
+            all_facts
+                .universal_region
+                .iter()
+                .map(|&origin| (origin, ())),
+        );
+
+        // TODO: flip the relation's arguments order in the input facts ?
```

Comment:
yeah, we might as well make it match what's most useful

Reply:
Turns out we use both orders:
- computing `known_contains` uses the origin key to join with `known_subset`
- computing `subset_errors` uses the placeholder loan as a key: the join will return the "source placeholder" associated to a loan which we find flowing in a downstream origin

The comment was correct earlier in the PR's life but not anymore, I'll remove it.